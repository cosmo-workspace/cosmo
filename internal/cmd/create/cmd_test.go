package create

import (
	"bytes"
	"testing"

	"github.com/cosmo-workspace/cosmo/pkg/cli"
	. "github.com/cosmo-workspace/cosmo/pkg/snap"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

	"github.com/spf13/cobra"
)

func TestCommandCreate(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "cosmoctl create suite")
}

var _ = Describe("help", func() {
	It("should match snapshot", func() {
		cmd := &cobra.Command{}
		out := bytes.Buffer{}
		cmd.SetOut(&out)
		AddCommand(cmd, cli.NewRootOptions())
		cmd.SetArgs([]string{"create", "--help"})
		err := cmd.Execute()
		Expect(err).ShouldNot(HaveOccurred())
		Expect(out.String()).To(MatchSnapShot())
	})
})
