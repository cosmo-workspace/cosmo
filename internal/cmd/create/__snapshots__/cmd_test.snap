['help should match snapshot 1']
SnapShot = """
Create cosmo resources

Usage:
   create [command]

Available Commands:
  network     Upsert workspace network. Alias of 'cosmoctl workspace upsert-network'
  user        Create user. Alias of 'cosmoctl user create'
  workspace   Create workspace. Alias of 'cosmoctl workspace create'

Flags:
  -h, --help   help for create

Use \" create [command] --help\" for more information about a command.
"""
